### DISCUSSION TITLE
execCommand() is deprecated. What will TinyMCE do?

### DISCUSSION THREAD

**Discussion by stevenssz** [tinymce/tinymce#7718] [General]
[It looks like execCommand() is deprecated](https://developer.mozilla.org/en-US/docs/Web/API/Document/execCommand), and TinyMCE uses it extensively. Are there any plans to address this?

**Comment by TheSpyder** [comment]
This API has been deprecated for a long time. Many many websites depend on it, it's not going away anytime soon. Having said that the majority of the calls to `execCommand` within the TinyMCE codebase are handled internally not by the browser. We might look at renaming the method in a future release to clear up this confusion.

Our long-term plan to address this deprecation is the new editor model developed for real-time collaboration:
https://www.tiny.cloud/blog/real-time-collaborative-editing-slate-js/

Once the model is ready for general use it will be added to our open source offering and the plan is it will eventually become the default for TinyMCE.
